### PR TITLE
Add Testnet Rewards Section in Astral Explorer

### PR DESCRIPTION
**Pull request by marc-aurele-besner**
## Preview

Logout-TestnetRewardsSection

Login-1wallet-TestnetRewardsTop
Login-3wallets-TestnetRewardsTop

___

### **PR Type**
enhancement, tests

___

### **Description**
- Added a new `TestnetIcon` component with SVG icons for various testnet phases.
- Implemented the `RewardHistory` component to manage and display connected wallets, including functionality to connect and disconnect subspace wallets.
- Created a responsive `EmptyHeader` component with theme toggle and chain dropdown support.
- Enhanced the user interface with new icons and improved wallet management features.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>TestnetIcon.tsx</strong><dd><code>Add SVG icons for testnet phases</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/icons/TestnetIcon.tsx

<li>Added multiple SVG icons for various testnet phases.<br> <li> Defined React functional components for each icon.<br> <li> Used consistent styling and color themes for icons.

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-1c0f55365c6d1c0ef10fb8c470b4c786254435cd05f0b1ce4e71c50c95b68ecf">+316/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>RewardHistory.tsx</strong><dd><code>Implement RewardHistory component for wallet management</code>&nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/TestnetRewards/RewardHistory.tsx

<li>Implemented <code>RewardHistory</code> component to display connected wallets.<br> <li> Added functionality to connect and disconnect subspace wallets.<br> <li> Utilized hooks for managing wallet connections and view states.

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-4a2b1b7af0e4832ec03644484d8da29c346633b77d003d7df9f53880f488a26c">+133/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>EmptyHeader.tsx</strong><dd><code>Create responsive EmptyHeader with theme toggle</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/layout/EmptyHeader.tsx

<li>Created <code>EmptyHeader</code> component with responsive design.<br> <li> Integrated theme toggle functionality.<br> <li> Included chain dropdown and mobile header support.

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-24961969d853b37cf0dc04e62f145a745353cf7295f32dfda3415a6c45e08b2e">+83/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Additional files (token-limit)</strong></td><td><details><summary>13 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>testnetRewards.ts</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/gql/types/testnetRewards.ts

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-48ef3d8434e48617844bb47a7089c0989059f269fd55b63a1db1bf7af4ec0e94">+2283/-0</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>TestnetRewardsTable.tsx</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/TestnetRewards/TestnetRewardsTable.tsx

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-35e3074a14443524d99d3004b4fa803bb33ed43a0649f85383a66ac28bdc1e76">+251/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>testnetRewards.query.ts</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/TestnetRewards/testnetRewards.query.ts

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-1f3b73e6e091c8382c0317a8dac5d303e781261d410dfe9e22f371c968d5ed33">+79/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>FAQ.tsx</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/TestnetRewards/FAQ.tsx

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-9791cba25d8e1daac07fa92bc83870e6698cd77384bba3032e12345f35ac4f68">+45/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>view.ts</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/states/view.ts

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-5365bc92bb38e535090ab99368f4aaf5ad43cd459db3e71e4a1bbda4e49b35e0">+11/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>AccountPreviousRewards.tsx</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/Consensus/Account/AccountPreviousRewards.tsx

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-8e3727ff57139c5207fda22d91439d50e5316030f92ce46cf616395d4c6a25e2">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>indexers.ts</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/constants/indexers.ts

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-75684240196dd4e05e14dbed5acaa52b800290949f62acd6fa8142cf6a425607">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/app/[chain]/testnet-rewards/page.tsx

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-7a80781dd52a341f139dd29b92dbae062553fa14fedda5eac01405c2c9bf03df">+29/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>routes.ts</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/constants/routes.ts

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-1db40c1c0135e9c55c149a79c123aafb3fc2408f35965369fe6ff07598b37d91">+5/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>index.tsx</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/components/TestnetRewards/index.tsx

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-38ec147ea887f20a7ef75ecdda1e5d799057746097aee1b94ad9ed6bd286e293">+24/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ChainProvider.tsx</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/providers/ChainProvider.tsx

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-a55a9b593552c3724c0ea05e726001ec210930c00c73a6549093b80abe4f7ced">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>codegen.ts</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/codegen.ts

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-54f0d956307bff1a7987d1fa957c2f0dfc8ba1730b55474de11f487ae4804e09">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>layout.tsx</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

explorer/src/app/[chain]/testnet-rewards/layout.tsx

...

</details>

  </td>
  <td><a href="https://github.com/autonomys/astral/pull/850/files#diff-918cfcbd114c22746c6337e0383e0ec94c214cde8e9149b4487bac67535d493e">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by marc-aurele-besner** [comment]
Saw a bunch of merge error, let me fix it and finalize the PR first

**Review by clostao — DISMISSED** [review]
LGTM

**Comment by marc-aurele-besner** [comment]
Aries Stress Test Allocation has been added

File: explorer/src/components/TestnetRewards/TestnetRewardsTable.tsx [hunk 1]
```diff
@@ -172,93 +200,101 @@ export const TestnetRewardsTable: FC = () => {
   const cleanCell = (cell: string) => cell.replace(/"/g, '').replace(/,/g, '')
 
   const handleLoad = useCallback(async () => {
-    const file = await fetch('/data/testnet/astralTestnetRewardsMainnetAllocations-10oct2024.csv')
+    const file = await fetch('/data/testnet/astralTestnetRewardsMainnetAllocations-10oct2024-2.csv')
```
Comment (clostao) [inline explorer/src/components/TestnetRewards/TestnetRewardsTable.tsx:203]: Why is this file `-2` instead of substituting?

**Review by clostao — APPROVED** [review]
LGTM

File: explorer/src/components/TestnetRewards/TestnetRewardsTable.tsx (hunk 1 above)
Comment (marc-aurele-besner) [reply explorer/src/components/TestnetRewards/TestnetRewardsTable.tsx:203]: Will substitute them on the next version (still waiting for final confirmation on Aries allocation)

**Review by clostao — DISMISSED** [review]
Awesome

**Review by clostao — APPROVED** [review]

**Pull request merged**